### ISSUE TITLE
Related to #11287: Bitwarden Safari Plugin don't stop open tabs

### ISSUE THREAD

**Issue by aamoh22** [bitwarden/clients#11694]
### Steps To Reproduce

Steps To Reproduce:

1. Install the Bitwarden macOS client from the App Store.
2. Activate the Safari Plugin from Safari's Extensions preferences.
3. Observe that the plugin intermittently opens new tabs displaying the "Getting started" page on bitwarden.com.
4. The issue occurs when the user is not logged in to Bitwarden, but also when they are not interacting with Safari at all.
5. After manually closing the tabs, they reappear after a while, seemingly every 1-2 hours.

### Expected Result

Bitwarden should not repeatedly open the "Getting started" page in new Safari tabs.

### Actual Result

The Safari extension continuously opens new tabs showing the "Getting started" page, even if such a tab is already open or was previously opened.
This behavior causes an overflow of tabs in Safari, requiring manual effort to close them.

### Operating System

macOS

### Operating System Version

15.0.1

### Web Browser

Safari

### Browser Version

18.0.1 

### Build Version

2024.9.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by jtodddd** [comment]
Hi there,

I attempted to reproduce your issue and was unable to do so. 

We use GitHub issues as a place to track bugs and other development related issues. If your issue persists, please write us back using our “Contact support” form located on our Help Center (https://bitwarden.com/help/).

You can include a link to this issue in the message content.

Alternatively, you can also search for an answer in our help documentation or get help from other Bitwarden users on our community forums (https://community.bitwarden.com/c/support/). 

The issue here will be closed.

Thanks!

**Comment by aamoh22** [comment]
Hi,

Thank you for your response. However, the issue persists despite Bitwarden marking it as resolved in version 2024.9.1 of the Safari extension, as noted in issues #11287 and #11356. I’ve followed the instructions and confirmed the extension is up-to-date, yet the problem remains.

I appreciate your suggestion to contact support via the Help Center, but we have already tried that, and unfortunately, the response was not helpful. Other users have reported similar issues in the same thread, suggesting that this may not be an isolated case.

Please escalate this matter, as it’s critical for us to have this resolved. We're happy to provide any additional information needed to assist in troubleshooting.

Looking forward to your prompt attention to this matter.

From: jtbw ***@***.***>
Date: Monday, October 28, 2024 at 9:41 AM
To: bitwarden/clients ***@***.***>
Cc: Ahmed Mohamed ***@***.***>, Author ***@***.***>
Subject: Re: [bitwarden/clients] Related to #11287: Bitwarden Safari Plugin don't stop open tabs (Issue #11694)


Hi there,

I attempted to reproduce your issue and was unable to do so.

We use GitHub issues as a place to track bugs and other development related issues. If your issue persists, please write us back using our “Contact support” form located on our Help Center (https://bitwarden.com/help/).

You can include a link to this issue in the message content.

Alternatively, you can also search for an answer in our help documentation or get help from other Bitwarden users on our community forums (https://community.bitwarden.com/c/support/).

The issue here will be closed.

Thanks!

**Comment by petarhristov94** [comment]
I'm experiencing an issue where Bitwarden opens multiple tabs to the "_Getting Started_" page (`https://bitwarden.com/browser-start/`) after unlocking the vault. The timing between unlocking and the tabs opening varies - sometimes immediately, other times after a few minutes.

Recently, I had two Safari windows open on different profiles, and suddenly 3 tabs opened in one window and another 3 tabs in the second window, forcing me to close 6 tabs in total. This behavior seems random and disruptive.

I’ve tried reproducing the issue, and it consistently happens the first time I open the Bitwarden desktop app (**Version 2024.11.0 (33157)**) on **macOS Sequoia 15.1.1 (24B91)**, though _I experienced the same problem on the previous macOS version as well_.

Interestingly, I also use the Bitwarden desktop client on a Windows system, and I have never encountered this issue there. In fact, I’ve never seen the "_Getting Started_" page open on my Windows system at all.

I’d appreciate guidance on how to disable this behavior entirely or any steps to resolve this issue. It seems like a bug specific to macOS or the Safari extension.

**Comment by petarhristov94** [comment]
I'm experiencing an issue where Bitwarden opens multiple tabs to the "_Getting Started_" page (`https://bitwarden.com/browser-start/`) after unlocking the vault. The timing between unlocking and the tabs opening varies - sometimes immediately, other times after a few minutes.

Recently, I had two Safari windows open on different profiles, and suddenly 3 tabs opened in one window and another 3 tabs in the second window, forcing me to close 6 tabs in total. This behavior seems random and disruptive.

I’ve tried reproducing the issue, and it consistently happens the first time I open the Bitwarden desktop app (**Version 2024.11.0 (33157)**) on **macOS Sequoia 15.1.1 (24B91)**, though _I experienced the same problem on the previous macOS version as well_.

Interestingly, I also use the Bitwarden desktop client on a Windows system, and I have never encountered this issue there. In fact, I’ve never seen the "_Getting Started_" page open on my Windows system at all.

I’d appreciate guidance on how to disable this behavior entirely or any steps to resolve this issue. It seems like a bug specific to macOS or the Safari extension.

**Comment by nicolas-cherel** [comment]
I've this issue at each time bitwarden start, https://bitwarden.com/browser-start/ is opened on safari and *each web app* (applications "added to dock" with an independent browser instance)

This means that I sometimes have 4-5 very unwanted tabs / new windows opened automatically.

I understand that detecting if users have completely setup their configuration to have the target experience is a hard problem, but allowing false negative instead of false positives at the expense of existing, paying customer does seem to be a reasonable tradeoff.

**Comment by addisonbeck** [comment]
I'm working on troubleshooting this now. Can anyone on this thread confirm some configuration settings for me?

I'm currently thinking the issue is related to the native messaging needed to facilitate touch id in the browser extension.

Can anyone who experiences this that _does not use touch id_ confirm that "Unlock with touch id" is disabled in the desktop app and browser extension?

**Comment by georgesavill** [comment]
> I'm working on troubleshooting this now. Can anyone on this thread confirm some configuration settings for me? I'm currently thinking the issue is related to the native messaging needed to facilitate touch id in the browser extension. Can anyone who experiences this that _does not use touch id_ confirm that "Unlock with touch id" is disabled in the desktop app and browser extension?

I can't see an option for touch ID anywhere, but I've got both biometrics and pin unlock disabled. I've just tried deleting all of the extension data, uninstalled it, reinstalled it, and set up connection to my vault - so I assume i'm running all the default settings. I'm still getting the tabs opening on every new browser session (i.e. first browser window opened for a particular profile between each reboot).

I'm connecting to a self-hosted vault, running Safari 18.2 on an M2 macbook air.

**Comment by infinitepower18** [comment]
After today's update it opened an endless number of Getting Started tabs. I'm sorry but it's crazy for a browser extension to spawn this many tabs. Please fix this ASAP.

![Image](https://github.com/user-attachments/assets/ef546a21-2136-476f-8715-b0651f54af9d)

**Comment by RiRilot** [comment]
Same for me yesterday evening. About 20 tabs spawned. Each time I closed them another few appeared.
This is making me look at alternative to Bitwarden now.

**Comment by addisonbeck** [comment]
The fact that these latest reports all talk this issue coming up when _updating_ is helpful. Thanks folks!

**Comment by LeTak0** [comment]
As far as I noticed it on my side. For each update that the extension had, it opens one tab, so if I open a safari profile after 5 bitwarden updates, there will be 5 new getting started tabs.
A checkbox in the settings to block/deactivate those tabs would be helpful.

**Comment by addisonbeck** [comment]
I'm considering doing that automatically - saving some bit to local storage saying "I installed the extension", and skipping the getting started screen if that is found.

It feels hacky, but that might be the direction this goes in. I haven't made it back to this yet since the most recent reports but will be in the next couple of days.

**Comment by Nevarro** [comment]
It's really annoying. Please fix this issue.

**Comment by Axynokay** [comment]
> Thank you for reporting this issue! We've added this to our internal tracking system. ID: PM-14012

Please fix this, or my clients will start looking for a different password manager. It’s becoming extremely annoying!

**Comment by Nexulo** [comment]
When I switch to a different browser profile in Safari on macOS, it automatically opens a new tab to bitwarden.com every time - this has been annoying for years.

I suspect that this happens not only after every update, but whenever I start the browser and switch to a different browser profile for the first time.

**Comment by vinnymac** [comment]
I recommend Bitwarden add a boolean setting so it never under any circumstance ever opens in a new tab. A root cause analysis is not needed here. Just set it to false by default if you need this in the product, and leave it up to us (the user) if we want to explicitly disable this behavior.

If you ever do fix this issue, great, but I don't think that's what anyone cares about in this thread.

**Comment by diamondsw** [comment]
While I agree with the tactical solution (save a bool saying never do this), I have some concerns that the root cause is something in Safari preventing Bitwarden from saving its settings, so any attempt to save state - even the boolean - might not work. I've sometimes seen other settings reset spontaneously (e.g. autofill settings) - could be unrelated, but seemed to start at the same time. Worth a shot, of course, but I wouldn't get my hopes up.

**Comment by vinnymac** [comment]
That sounds like a critical bug in this extension if it exists, but regardless, you don't have to use the existing store (that may be buggy) to fix this one issue. Which API is actively in use?

There are a plethora of locations where a value can be stored to alleviate the issue described above.

- an external file store (s3 compatible API)
- an external cache or database
- local indexed db
- the [File System Access API](https://webkit.org/blog/12257/the-file-system-access-api-with-origin-private-file-system/)

Even a TXT dns record could be used, and many more, but I think you get the point. You can implement this so the setting is stored in multiple places with redundancies.

**Comment by nicolas-cherel** [comment]
Came back to three undesirable tabs this morning. Three per safari instance, since i'm using desktop web app. 

![Image](https://github.com/user-attachments/assets/667e0cee-d0e2-4006-aeea-c23f8a436a66)

**Comment by diamondsw** [comment]
@addisonbeck Sure, I'd be happy to. I just migrated to a new computer, so between that and these updates, I'll let the dust settle a bit before filing the issue. Since no one else seems to have said issue, it's highly likely it's a Just Me problem.

**Comment by Axynokay** [comment]
I also have this issue! But what I don’t understand is why they wouldn’t just shut off this «feature» completely. Why does it need to open a tab with the Bitwarden website like an ad?

**Comment by Nexulo** [comment]
I don't understand that either. It's not as if anyone couldn't simply find the Bitwarden website themselves and get information. That's why the reasoning is incomprehensible to me. 

Users can really inform themselves the first time they use it and are certainly happy to do without automatically opening tabs at every start. For me, it always opens the Bitwarden page when I change the Safari profile. Sure, I can click it away, but that is annoying.

**Comment by Axynokay** [comment]
It gets even worse. As an IT service provider, I manage clients who use multiple Mac accounts with the same Apple account. Each team member has their own Safari profile, but since the Bitwarden plugin for Safari syncs via the same iCloud account, disabling and re-enabling it causes the Bitwarden site to pop up on all Macs, confusing everyone and leading to support tickets.

Turn this nonsense off! No user needs this Bitwarden ad. Why? We’re already Bitwarden users!

Once again, the marketing department has outdone itself. Bravo! /s

**Comment by moussak4** [comment]
> [@addisonbeck](https://github.com/addisonbeck) Sure, I'd be happy to. I just migrated to a new computer, so between that and these updates, I'll let the dust settle a bit before filing the issue. Since no one else seems to have said issue, it's highly likely it's a Just Me problem.

Not a Just You problem. I have the same issue with the "Show autofill suggestions on form fields" setting: I keep disabling it, but it keeps re-enabling itself after some time. It's also specific to Safari profiles for me, I don't have any issue with the default profile (both the Getting Started page issue, and the "setting doesn't save" issue).

**Comment by diamondsw** [comment]
> > [@addisonbeck](https://github.com/addisonbeck) Sure, I'd be happy to. I just migrated to a new computer, so between that and these updates, I'll let the dust settle a bit before filing the issue. Since no one else seems to have said issue, it's highly likely it's a Just Me problem.
> 
> I have the same issue with the "Show autofill suggestions on form fields" setting: I keep disabling it, but it keeps re-enabling itself after some time. It's also specific to Safari profiles for me, I don't have any issue with the default profile (both the Getting Started page issue, and the "setting doesn't save" issue).

That's it exactly! I don't know if other settings are impacted as well, but that's what I keep changing constantly, and keeps getting reset.

**Comment by addisonbeck** [comment]
A fix has been released that resolves the original issue from this thread: the getting started paging opening when it shouldn't in Safari. This has been out for a bit now, and most folks should have updated out of seeing the problem. If you still see this happening please make sure you're on the latest version.

The issues being reported about fields not sticking is separate, and I'd ask that anyone interested please open a new issue for that problem so we can diagnose and track it separately.

**Comment by flostellbrink** [comment]
I got the welcome page for all of my Safari profiles again today. I am on 2025.5.1.
